tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -292,10 +315,38 @@ public String format(NameObject nameObject) {
                 if (literal != null) {
                     result.append(literal);
                 } else {
-                    final String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
-                    if (!remainingModifers.isEmpty()) {
+                    String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
+                    if (bestValue == null) {
+                        // TODO Rich to handle collapsing whitespace / literals
```

Comment:
I'll try to do this on Monday; I ran out of time today.

Reply:
Thanks!